### PR TITLE
competition-centernetv2

### PR DESCRIPTION
**Pull request by huimlight**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Learn Centernetv2, learn MMDET.

## Modification

Code: we add detector in mmdet/models/detectors/centernetv2.py.

Code: we add config in configs/centernetv2/centernet2.py.

Code: we add centernet_head in mmdet/dense_heads/centernet_headv2.py.

Code: we add hm_binary_focal_loss in mmdet/losses/hm_binary_focal_loss.py.

Code: we modify mmdet/roi_heads/cascade_roi_head.py, add the super parameter add_agnostic_score to Control whether the first stage score.This hyperparameter does not affect the use of other configuration files

Code: we modify mmdet/bbox_head/bbox_head.py, add the super parameter add_agnostic_score to Control Whether to use softmax.This hyperparameter does not affect the use of other configuration files

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
No

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by LMerCy** [comment]
@huimlight So you don't multiply  rpn score with rcnn score when training?

**Comment by huimlight** [comment]
> @huimlight So you don't multiply rpn score with rcnn score when training?

no, I multiply. This code in cascade_roi_head.py

**Comment by FDInSky** [comment]
很棒的工作，什么时候可以合入啊？

**Comment by wusize** [comment]
> @huimlight So you don't multiply rpn score with rcnn score when training?

It's not multiplied in training based on what I see. In fact, the rpn score is not used to calculate rcnn loss in the [official implemenation](https://github.com/xingyizhou/CenterNet2/issues/12).

**Comment by wusize** [comment]
> @huimlight So you don't multiply rpn score with rcnn score when training?

It's not multiplied in training based on what I see. In fact, the rpn score is not used to calculate rcnn loss in the [official implemenation](https://github.com/xingyizhou/CenterNet2/issues/12).

File: mmdet/models/dense_heads/centernet_headv2.py [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+    def _init_layers(self):
+        """Initialize layers of the head."""
+        self._init_reg_convs()
+        self._init_predictor()
+        self.scales = nn.ModuleList([Scale(1.0) for _ in self.strides])
+
+    def _init_predictor(self):
+        """Initialize predictor layers of the head."""
+        self.conv_reg = nn.Conv2d(self.feat_channels, 4, 3, padding=1)
+        self.agn_hm = nn.Conv2d(self.in_channels, 1, 3, stride=1, padding=1)
```
Comment (BIGWangYuDong) [inline mmdet/models/dense_heads/centernet_headv2.py:96]: Seems did not use num_classes? should we set num_classes=1 in config file?

**Comment by ZwwWayne** [comment]
closed as it has been continued in https://github.com/open-mmlab/mmdetection/pull/9566
